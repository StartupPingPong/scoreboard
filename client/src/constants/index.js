// titles, subtitles, brands, etc
export const strings = {
  eventName: "Startup Ping-Pong"
};
