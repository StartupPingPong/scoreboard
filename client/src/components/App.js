import React, { Component } from "react";
import { connect } from "react-redux";
import { strings } from "../constants";
import * as actions from "../actions";

export class App extends Component {
  componentDidMount() {
    this.props.dispatch({ type: actions.GET_ALL_TEAMS });
  }

  render() {
    const { teams } = this.props;
    return (
      <div>
        <h1 className="main-title display-1">{strings.eventName}</h1>
        {teams.map(team => (
          <p className="lead" key={team.id}>
            {team.teamName}
          </p>
        ))}
      </div>
    );
  }
}

export default connect(state => ({ teams: state.teams }), null)(App);
