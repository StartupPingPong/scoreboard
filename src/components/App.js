import React, { Component } from "react";
import { strings } from "../constants";

class App extends Component {
  render() {
    return (
      <div>
        <h1 className="main-title display-1">{strings.eventName}</h1>
      </div>
    );
  }
}

export default App;
