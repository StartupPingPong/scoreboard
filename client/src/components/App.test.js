import React from "react";
import { shallow } from "enzyme";

import { App } from "./App";
import { strings } from "../constants";

describe("App", () => {
  let props = {
    teams: [
      {
        id: 0,
        teamName: "test me"
      }
    ]
  };

  let app = shallow(<App {...props} />);

  it("renders properly", () => {
    expect(app).toMatchSnapshot();
  });

  let title = app.find(".main-title");
  it("displays `SUPP` title", () => {
    expect(title.exists()).toBe(true);
    expect(title.text()).toEqual(strings.eventName);
  });

  it("renders the teams given as props", () => {
    expect(app.find(".lead").text()).toEqual("test me");
  });
});
