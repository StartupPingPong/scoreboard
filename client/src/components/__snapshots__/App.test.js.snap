// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    teams={
        Array [
            Object {
              "id": 0,
              "teamName": "test me",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-title display-1"
>
          Startup Ping-Pong
</h1>,
        Array [
          <p
            className="lead"
>
            test me
</p>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Startup Ping-Pong",
          "className": "main-title display-1",
        },
        "ref": null,
        "rendered": "Startup Ping-Pong",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "test me",
          "className": "lead",
        },
        "ref": null,
        "rendered": "test me",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-title display-1"
>
            Startup Ping-Pong
</h1>,
          Array [
            <p
              className="lead"
>
              test me
</p>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Startup Ping-Pong",
            "className": "main-title display-1",
          },
          "ref": null,
          "rendered": "Startup Ping-Pong",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "test me",
            "className": "lead",
          },
          "ref": null,
          "rendered": "test me",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
